tree-sitter-stack-graphs/src/cli/util/reporter.rs

```diff
@@ -0,0 +1,178 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2023, stack-graphs authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+use colored::ColoredString;
+use colored::Colorize;
+use std::io::Write;
+use std::path::Path;
+
+/// Trait that supports reporting file processing status.
+///
+/// For each file, either
+///  - [`skipped`] is called once, or
+///  - [`started`] and one of [`succeeded`], [`failed`], or [`canceled`] are called.
```

Comment:
"and one" ⇒ "and then one" to describe the required ordering of calls

Reply:
Yep, that's better!